packages/cloudflare/src/cli/args.ts

```diff
@@ -81,9 +84,26 @@ export function getArgs(): Arguments {
   }
 }
 
-function getPassthroughArgs() {
-  const passthroughPos = process.argv.indexOf("--");
-  return passthroughPos === -1 ? [] : process.argv.slice(passthroughPos + 1);
+export function getPassthroughArgs<T extends ParseArgsConfig>(args: string[], { options = {} }: T) {
+  const passthroughArgs: string[] = [];
+
+  for (let i = 0; i < args.length; i++) {
+    if (args[i] === "--") {
+      passthroughArgs.push(...args.slice(i + 1));
+      return passthroughArgs;
+    }
+
+    const [, name] = /^--?(\w[\w-_]*)(=.+)?$/.exec(args[i]!) ?? [];
+    if (name && !(name in options)) {
+      passthroughArgs.push(args[i]!);
+
+      while (!args[i + 1]?.startsWith("-")) {
```

Comment:
Thanks for the update. 
I find the new version much more readable.

Still wondering why we have a while here.

I get the `--flagName flagValue` thing but why more strings after `flagValue` ?